src/user/user.routes.ts

```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import { validateDto } from '../middleware/validateDto';
+import { changeInfoDto } from '../auth/dto/changeInfo-dto';
+import { protect } from '../middleware/protect'
+
+const userRoutes = express.Router();
+
+userRoutes.patch('/api/users', protect, validateDto(changeInfoDto),)
```

Comment:
You have not actually implemented the logic to change the user info, you have the controller, with the endpoint, protect, validateDto, and you have to create a handler function for this endpoint in a seperate file ex: "user.handler.ts" which would have a changeUserInfo exported function, that actually changes the user info in the database.

I Also have misled you in the task description, the endpoint should be "/api/users/:id", the ":id" represents user id, so when we will want to change the info of a user with id 3, we will send the request to "/api/users/3", if id is 5: "/api/users/5"

also in the user.routes.ts file the endpoint should only be "/:id" and in the app.ts file you should have:
app.use('/api/users, userRoutes); see example of how it was done with authRoutes